lib/functions/name/name.ts

```diff
@@ -0,0 +1,46 @@
+interface FormatDisplayNameInput {
+  /** Full name of person or company */
+  fullName: string;
+  /** Entity type: Whether this is a person or a company */
+  type: 'person' | 'company';
+  /** For persons: switch order of first/last name */
+  reverseNameOrder?: boolean;
+}
+
+/**
+ * Formats a display name for a person or company by applying title casing,
+ * normalizing hyphenated words, and (for companies) uppercasing known legal
+ * entity suffixes.
+ *
+ * For persons, you can optionally reorder the name by moving the first word
+ * (often a surname) to the end.
+ *
+ * @param {FormatDisplayNameInput} input - The input object with full name, entity type, and options.
+ * @returns {string} The formatted display name.
+ *
+ */
+export const formatDisplayName = ({ fullName, type, reverseNameOrder }: FormatDisplayNameInput): string => {
+  if (!fullName) return '';
+
+  const legalEntityTypes = ['enk', 'as', 'da', 'sa'];
```

Comment:
```suggestion
  const legalEntityTypes = ['enk', 'as', 'da', 'sa', 'asa', 'ba', 'ans', 'sti', 'nuf' ];
```
I asked Remi to do a check for the most common such types in names in production, and this is that list. (there are others, but they have less than 1 500 cases of occurrence so I capped it at that. I do have the full list though, if we want them all)